Volto docs integration (#1160)
plone/documentation: docs/contributing/index.md

Quick edits for minor issues, such as typographical errors, misspellings, and English grammar and syntax, can be made through the GitHub user interface.

1.  Navigate to the repository as noted in {ref}`contributing-manage-on-github-label`.
1.  Navigate with the `docs` directory to find the source file to edit.
1.  Click the {guilabel}`pencil icon` to edit the file in the browser.

    ```{image} /_static/edit-pencil-icon.png
    :alt: GitHub Edit this file
    ```
1.  Make edits, add a commit message, select {guilabel}`Create a **new branch** for this commit and start a pull request`, then click {guilabel}`Propose changes`.
1.  Make your pull request against the branch `6-dev`.
1.  Members who subscribe to the repository will receive a notification and review your request.

(contributing-large-edits-label)=

### Large edits

For large edits, first follow the instructions in {doc}`setup-build`.

Once you have your environment set up, then you can follow the standard practice for making a pull request.
This practice differs depending on whether you are making contributions to only the core `documentation` files or `volto` files as well.

(contributing-documentation-only-label)=

> ### Working with only the `documentation` repository

Reviewer: ```suggestion
### Working with the `plone/documentation` repository
```
- explicit mention of plone/documentation in contrary to plone/volto/docs
- request to native speakers: where and if to put the  word 'only'?

Reviewer 2: Fixed in e8d1338865d54b825addfe54761b10285d866628

The intent is to say that the contributor is working with only this repo and no other.

"Only" is a fascinating English word. It gets tossed around for emphasis and meaning. It's hard to explain. Here's a good example.

https://englishwithnab.com/place-the-word-only-anywhere-on-the-sentence/

> This section describes how to make contributions to files in the `documentation` repository only, and excludes `volto` files.

Reviewer: ```suggestion
This section describes how to make contributions to files in the `plone/documentation` repository only, and excludes `plone/volto/docs` files.
```

> 1.  From the project root directory, synch your local `6-dev` branch with its remote.

Reviewer: ```suggestion
1.  From the project root directory, sync your local `6-dev` branch with its remote.
```

Reviewer 2: I am showing my age. Trends over the last two decades show that "sync" is now the more common usage. Reference: https://grammarhow.com/sync-vs-synch/

    You might need to resolve conflicts.

    ```shell
    git checkout 6-dev
    git pull
    ```

1.  Create a new branch from `6-dev`.

    ```shell
    git checkout -b <new_branch>
    ```

1.  Edit files, save, preview, and test.
    You must run and pass the builds `html` and `linkcheck` without causing new errors.

    ```shell
    # Optionally clean the builds to avoid cache issues
    make clean
    make html
    make linkcheck
    ```

    ```{note}
    Currently there are some errors on the `html` build, mostly due to empty `meta` HTML tags.
    You are welcome to fix as many errors as you like.
    You are only responsible to fix errors that you create.
    ```

    ```{note}
    Eventually the `spellcheck` build will be required, but not at this time.
    We welcome improvements to the dictionary.
    ```

    ```{seealso}
    {ref}`setup-build-available-documentation-builds-label`.
    ```

> 1.  After the builds pass, commit changes to your branch, and push it to the remote repository on GitHub.

Reviewer: ```suggestion
1.  After the build pass, commit changes to your branch, and push it to the remote repository on GitHub.
```

Reviewer 2: There are multiple builds that must pass, not just one.

    The remote repository may be either a branch in your fork of the project or a branch in the `plone/documentation` upstream repository.

    ```shell
    git commit -m "My descriptive commit message"
    git push
    ```

1.  Visit the GitHub `documentation` repository, and [create a pull request](https://github.com/plone/documentation/compare) against the branch `6-dev`.
1.  Members who subscribe to the repository will receive a notification and review your request.

(contributing-documentation-and-volto-label)=

> ### Working with both the `documentation` and `volto` repositories

Reviewer: ```suggestion
### Editing Volto documentation

Working with both the `documentation` and `volto` repositories
```

Reviewer 2: Fixed in 89e81f9cddbd899687c0d9eaf2e6118963d076c4

> When you want to edit files within the `docs/volto` subdirectory, the process is slightly different.

Reviewer: ```suggestion
If you want to edit the Volto documentation: files within the `docs/volto` subdirectory, the process is slightly different.
```

Reviewer 2: There is an English phrase, "It's not 'if', but 'when`," meaning that something is inevitable. You're right, it is not inevitable, but it is from only my perspective 😆 .

Fixed in 32fb1ba43814949f76148cc9e6cafee726b8834a and 38c9510bd181045bee7592dec577fcf38e926ef4

We use git submodules to manage multiple repositories.

> We imported the `volto` repository into the `documentation` repository as described in {doc}`setup-build`.

Reviewer: It is not described in the mentioned section.

Reviewer 2: @ksuess I rewrote this passage. Also see in `setup-build`:

> Then with a single command using `Makefile`, create a Python virtual environment, install project dependencies, **pull in Volto documentation as a git submodule**, build the docs, and view the results in a web browser by opening `/_build/html/index.html`.

```{important}

> We currently use the branches `plone/documentation@6-dev` and `plone/volto@plone6-docs` as the main branches for developing Plnoe 6 Documentation.

Reviewer: ```suggestion
We currently use the branches `plone/documentation@6-dev` and `plone/volto@plone6-docs` as the main branches for developing Plone 6 Documentation.
```

These branches may change as we get closer to a production release.
```

> 1.  From the project root directory, synch your local `6-dev` branch with its remote.

Reviewer: native speaker: isn't it sync?

Reviewer 2: Fixed in e45ae4f067067f407d894cd72efce49ed5b50f62

    You might need to resolve conflicts.

    ```shell
    git checkout 6-dev
    git pull
    ```

1.  Change your working directory from the project root directory to `submodules/volto`.

    ```shell
    cd submodules/volto
    ```

1.  Update the submodule, and synch your local `plone6-docs` branch with its remote.
    You might need to resolve conflicts.

    ```shell
    git submodule update
    git checkout plone6-docs
    git pull
    ```

1.  Create a new branch from `volto/plone6-docs`.

    ```shell
    git checkout -b <new_branch>
    ```

1.  Make edits to files in `docs/volto` using your favorite editor, and save, preview, and test.
    You must run and pass the builds `html` and `linkcheck` without causing new errors.

    ```shell
    # Optionally clean the builds to avoid cache issues
    make clean
    make html
    make linkcheck
    ```

> 

Reviewer: Add a changelog entry.

why not introduce  ## Documentation section in plone/volto?

```
### Documentation

- Add metadata on i18n page @janedoe
```

Reviewer 2: Agreed. I'll just move things around, and if anyone complains, slap me with a fish.

1.  Back in `submodules/volto`, commit and push your changes to the remote.

    ```shell
    git add <files>
    git commit -m "My commit message"
    git push
    ```

> 1.  Now return to the project root directory, and update the submodule to point to the commit you just made, and push your changes to the remote repository.

Reviewer: ```suggestion
1.  Now return to the project root directory, and update the submodule to point to the commit you just made, and push your changes to the remote /plone/documentation repository.
```

Reviewer 2: The remote branch could be in either a user's fork or the upstream repo `/plone/documentation`. Not being explicit allows either usage.

    ```shell
    cd ../..
    git add submodules/volto
    git commit -m "Update submodules/volto tip"
    git push
    ```

> 1.  Visit the GitHub `volto` repository, and [create a pull request](https://github.com/plone/volto/compare) against the branch `plone6-docs`.

Reviewer: This link is OK. But a link to a compare with the branch would be even better. Anyone expert on this field?

Reviewer 2: Yes, I tried, but it is a chicken-and-egg thing. You cannot create a link to compare a branch that does not exist. This is the best default option I could come up with.

Anyway, here is an example of comparing `master` as the base against `plone6-docs`: https://github.com/plone/volto/compare/master...plone6-docs

> 1.  Members who subscribe to the repository will receive a notification and review your request.

Reviewer: ```suggestion
1.  Members who subscribe to the plone/volto repository will receive a notification and review your request.
```

Reviewer 2: Fixed in e06cdbb891c7417b07a076bd0d48c38724c0d160
